WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/ui/handlers/StartupUIThread.java

```diff
@@ -20,29 +20,30 @@
  */
 public class StartupUIThread implements Runnable {
 
-	private final StartUpHandler startUpHandler;
-
 	/** The preferences. */
 	private Preferences preferences;
 
+	/** Whether user has cancelled first registration. */
+	private boolean userRegistrationCancelled = false;
+
 	/** Constructor. */
-	public StartupUIThread(StartUpHandler startUpHandler,
-			Preferences preferences) {
-		this.startUpHandler = startUpHandler;
+	public StartupUIThread(Preferences preferences) {
 		this.preferences = preferences;
 	}
 
 	@Override
 	public void run() {
 		// the execution strategy is:
-		// (1) check user registration (2) wait until 1 is complete, or 1
-		// can be skipped (3) show workspace registration
+		// (1) check user registration (2) wait until user+project registration
+		// is completed, or 1 can be skipped (3) if user is already registered,
+		// check for workspace registration
 		if (WatchDogUtils.isEmpty(preferences.getUserid())) {
 			displayUserRegistrationWizard();
 		}
 
-		if (!WatchDogUtils.isEmpty(preferences.getUserid())) {
-			// In case the user aborted the preference dialog with cancel,
+		if (!WatchDogUtils.isEmpty(preferences.getUserid())
```

Comment:
I don't understand this condition anymore. I assume the comment is not up-to-date anymore? Why are you checking and modifying userRegistrationCancelled?